swap/src/cli/command.rs

```diff
@@ -72,6 +73,12 @@ pub enum Command {
     },
 }
 
+fn parse_monero_address(s: &str) -> Result<monero::Address, anyhow::Error> {
+    monero::Address::from_str(s)
+        .map_err(|err| anyhow!(err))
+        .context("Unable to parse the given monero address, make sure it is a valid address")
```

Comment:
Using `with_context` you could say something like:

```
failed to parse "foobar" as a monero address
```

to give the user a clearer error message